profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PprofAllocationEventExporter.java

```diff
@@ -0,0 +1,291 @@
+    LocationTable locationTable = profileData.locationTable;
+    StringTable stringTable = profileData.stringTable;
+
+    // XXX jdk mission control "TLAB allocations" view seems to use tlabSize as allocation size
+    long allocationSize = event.getLong("allocationSize");
+
+    Sample.Builder sample = Sample.newBuilder();
+    sample.addValue(allocationSize);
+    // XXX StackSerializer limits stack depth. Although I believe jfr also limits it should we
+    // attempt to limit the depth here too?
+    for (RecordedFrame frame : stackTrace.getFrames()) {
+      RecordedMethod method = frame.getMethod();
+      if (method == null) {
+        sample.addLocationId(locationTable.get("", "unknown.unknown", -1));
+      } else {
+        sample.addLocationId(
+            locationTable.get(
+                "", // file name is not known
+                method.getType().getName() + "." + method.getName(),
+                frame.getLineNumber()));
+      }
+    }
+    if (spanContext != null && spanContext.isValid()) {
+      Label.Builder traceIdLabel = Label.newBuilder();
+      traceIdLabel.setKey(stringTable.get("trace_id"));
+      traceIdLabel.setStr(stringTable.get(spanContext.getTraceId()));
+      sample.addLabel(traceIdLabel.build());
+
+      Label.Builder spanIdLabel = Label.newBuilder();
+      spanIdLabel.setKey(stringTable.get("span_id"));
+      spanIdLabel.setStr(stringTable.get(spanContext.getSpanId()));
+      sample.addLabel(spanIdLabel.build());
+
+      Label.Builder eventTypeLabel = Label.newBuilder();
+      eventTypeLabel.setKey(stringTable.get("event_type"));
+      eventTypeLabel.setStr(stringTable.get(event.getEventType().getName()));
+      sample.addLabel(eventTypeLabel.build());
+    }
+    // XXX are there any more attributes that should be added to the sample?
+    // XXX PlainTextAllocationEventExporter adds attributes from the used AllocationEventSampler
+
+    profile.addSample(sample);
+  }
+
+  private byte[] serializePprof() {
+    Profile profile = profileData.profile.build();
+    profileData = new ProfileData();
+
+    try {
+      ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
+      boolean gzip =
+          allocationDataFormat == AllocationDataFormat.PPROF_GZIP
+              || allocationDataFormat == AllocationDataFormat.PPROF_GZIP_BASE64;
+      try (OutputStream outputStream = gzip ? new GZIPOutputStream(byteStream) : byteStream) {
+        profile.writeTo(outputStream);
+      }
+      byte[] bytes = byteStream.toByteArray();
+      if (allocationDataFormat == AllocationDataFormat.PPROF_BASE64
+          || allocationDataFormat == AllocationDataFormat.PPROF_GZIP_BASE64) {
+        bytes = Base64.getEncoder().encode(bytes);
+      }
+      return bytes;
+    } catch (IOException exception) {
+      throw new IllegalStateException("Failed to serialize pprof", exception);
+    }
+  }
+
+  public static final AttributeKey<String> DATA_TYPE = stringKey("com.splunk.datatype");
```

Comment:
I suggest we be a bit more specific here and let's add `profiling` in between, like `com.splunk.profiler.datatype`. Same for encoding.

Reply:
changed to `com.splunk.profiling.data.type` and `com.splunk.profiling.data.format`